src/NuGet.Core/NuGet.Build.Tasks/GetReferenceNearestTargetFrameworkTask.cs

```diff
@@ -78,58 +80,89 @@ public override bool Execute()
 
             return !Log.HasLoggedErrors;
         }
 
-        private ITaskItem AssignNearestFrameworkForSingleReference(ITaskItem project, IList<NuGetFramework> currentProjectTargetFrameworks)
+        private ITaskItem AssignNearestFrameworkForSingleReference(
+            ITaskItem project,
+            NuGetFramework projectNuGetFramework,
+            IList<NuGetFramework> fallbackNuGetFrameworks,
+            MSBuildLogger logger)
         {
             var itemWithProperties = new TaskItem(project);
-            var targetFrameworks = project.GetMetadata(TARGET_FRAMEWORKS);
+            var referencedProjectFrameworkString = project.GetMetadata(TARGET_FRAMEWORKS);
+            var referencedProjectFile = project.GetMetadata(MSBUILD_SOURCE_PROJECT_FILE);
 
-            if (string.IsNullOrEmpty(targetFrameworks))
+            if (string.IsNullOrEmpty(referencedProjectFrameworkString))
             {
                 // No target frameworks set, nothing to do.
                 return itemWithProperties;
             }
 
-            var possibleTargetFrameworks = MSBuildStringUtility.Split(targetFrameworks);
+            var referencedProjectFrameworks = MSBuildStringUtility.Split(referencedProjectFrameworkString);
+
+            // try project framework
+            var nearestNuGetFramework = NuGetFrameworkUtility.GetNearest(referencedProjectFrameworks, projectNuGetFramework, NuGetFramework.Parse);
+            if (nearestNuGetFramework != null)
+            {
+                itemWithProperties.SetMetadata(NEAREST_TARGET_FRAMEWORK, nearestNuGetFramework);
+                return itemWithProperties;
+            }
 
-            foreach (var currentProjectTargetFramework in currentProjectTargetFrameworks)
+            // try project fallback frameworks
+            foreach (var currentProjectTargetFramework in fallbackNuGetFrameworks)
             {
-                var nearestNuGetFramework = NuGetFrameworkUtility.GetNearest(possibleTargetFrameworks, currentProjectTargetFramework, NuGetFramework.Parse);
+                nearestNuGetFramework = NuGetFrameworkUtility.GetNearest(referencedProjectFrameworks, currentProjectTargetFramework, NuGetFramework.Parse);
 
                 if (nearestNuGetFramework != null)
                 {
+                    var message = string.Format(CultureInfo.CurrentCulture,
+                        Strings.ImportsFallbackWarning,
+                        referencedProjectFile,
+                        currentProjectTargetFramework.DotNetFrameworkName,
+                        projectNuGetFramework.DotNetFrameworkName);
+
+                    var warning = RestoreLogMessage.CreateWarning(NuGetLogCode.NU1702, message);
+                    warning.LibraryId = referencedProjectFile;
+                    warning.ProjectPath = CurrentProjectName;
+
+                    // log NU1702 for ATF on project reference
+                    logger.Log(warning);
```

Comment:
Just to understand how are we skipping this warning when customer has no warn this? or will it be MSBuild who will take care of that?

Reply:
As long as this code is unique to this situation, MSBuild [now has](https://github.com/Microsoft/msbuild/pull/1928) a general-purpose property `MSBuildWarningsAsMessages` that can suppress messages by code.

Reply:
msbuild now has support for thhis - https://github.com/Microsoft/msbuild/pull/1928

`<MSBuildWarningsAsMessages>NU1702</MSBuildWarningsAsMessages>`